#! /bin/sh -e
## 10_no_cpp_comments.dpatch by Petter Reinholdtsen <pere@hungry.com>
##
## All lines beginning with `## DP:' are a description of the patch.
##
## DP: Avoid C++ comments (//) to get the code to compile with -ansi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch) patch $patch_opts -p0 < $0;;
       -unpatch) patch $patch_opts -p0 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0
@DPATCH@
Index: src/buildmap_area.c
===================================================================
RCS file: /cvsroot/roadmap/roadmap/src/buildmap_area.c,v
retrieving revision 1.1.1.1
diff -u -3 -p -u -r1.1.1.1 buildmap_area.c
--- src/buildmap_area.c	21 Feb 2003 20:07:10 -0000	1.1.1.1
+++ src/buildmap_area.c	22 Nov 2004 07:23:25 -0000
@@ -63,7 +63,7 @@ int buildmap_area_add
          RoadMapString fedirs,
          int tlid) {
 
-   return 0; // TBD
+   return 0; /* TBD */
 }
 
 
Index: src/buildmap_dbwrite.c
===================================================================
RCS file: /cvsroot/roadmap/roadmap/src/buildmap_dbwrite.c,v
retrieving revision 1.1.1.1
diff -u -3 -p -u -r1.1.1.1 buildmap_dbwrite.c
--- src/buildmap_dbwrite.c	21 Feb 2003 20:07:11 -0000	1.1.1.1
+++ src/buildmap_dbwrite.c	22 Nov 2004 07:23:25 -0000
@@ -121,7 +121,7 @@ static void buildmap_db_extend (int size
 static void buildmap_db_propagate (buildmap_db *parent, int size) {
 
    while (parent != NULL) {
-      parent->head->count = 1; // Just a convention, not really used.
+      parent->head->count = 1; /* Just a convention, not really used. */
       parent->head->size += size;
       parent = parent->parent;
    }
Index: src/buildmap_place.h
===================================================================
RCS file: /cvsroot/roadmap/roadmap/src/buildmap_place.h,v
retrieving revision 1.1
diff -u -3 -p -u -r1.1 buildmap_place.h
--- src/buildmap_place.h	8 Nov 2004 07:53:42 -0000	1.1
+++ src/buildmap_place.h	22 Nov 2004 07:23:25 -0000
@@ -30,8 +30,8 @@ void buildmap_place_initialize (void);
 int  buildmap_place_add (int name, int cfcc, int point);
 int  buildmap_place_get_sorted  (int place);
 int buildmap_place_get_name_sorted (int place);
-//static BuildMapPlace *buildmap_place_get_record (int place);
-//static BuildMapPlace *buildmap_place_get_record_sorted (int place);
+/* static BuildMapPlace *buildmap_place_get_record (int place); */
+/* static BuildMapPlace *buildmap_place_get_record_sorted (int place); */
 int buildmap_place_find_sorted (int name);
 void buildmap_place_get_position 
         (int place, int *longitude, int *latitude);
@@ -39,11 +39,11 @@ int buildmap_place_get_point_sorted (int
 void buildmap_place_get_position_sorted
         (int place, int *longitude, int *latitude);
 int buildmap_place_get_square_sorted (int place);
-//int buildmap_place_compare (const void *r1, const void *r2);
+/* int buildmap_place_compare (const void *r1, const void *r2); */
 void buildmap_place_sort (void);
 void buildmap_place_save    (void);
 void buildmap_place_summary (void);
 void buildmap_place_reset   (void);
 
-#endif // _BUILDMAP_PLACE__H_
+#endif /* _BUILDMAP_PLACE__H_ */
 
Index: src/buildplace_main.c
===================================================================
RCS file: /cvsroot/roadmap/roadmap/src/buildplace_main.c,v
retrieving revision 1.1
diff -u -3 -p -u -r1.1 buildplace_main.c
--- src/buildplace_main.c	8 Nov 2004 07:53:42 -0000	1.1
+++ src/buildplace_main.c	22 Nov 2004 07:23:25 -0000
@@ -42,14 +42,14 @@
 
 #include <shapefil.h>
 
-// shapefile column names
+/* shapefile column names */
 
 #define F_NAME  "NAME"
 #define F_STATE "STATE"
 #define F_CC    "CC"
 #define F_TYPE  "TYPE"
 
-#endif // ROADMAP_USE_SHAPEFILES
+#endif /* ROADMAP_USE_SHAPEFILES */
 
 
 #define BUILDPLACE_FORMAT_SHAPE     1
@@ -287,7 +287,7 @@ static void buildplace_shapefile_process
         if (cfcc == 0)
             continue;
 
-        // add the place
+        /* add the place */
 
         shp = SHPReadObject(hSHP, irec);
 
@@ -318,7 +318,7 @@ static void buildplace_shapefile_process
             source);
    exit(1);
 
-#endif // ROADMAP_USE_SHAPEFILES
+#endif /* ROADMAP_USE_SHAPEFILES */
 }
 
 
@@ -352,7 +352,7 @@ static void buildplace_read_dsg (const c
             if (cfcc < 0 || cfcc > BUILDMAP_MAX_PLACE_CFCC)
                 buildmap_fatal (0, "place cfcc is out of range");
             
-            while (isspace(*p)) p++; // skip leading blanks
+            while (isspace(*p)) p++; /* skip leading blanks */
             c = strcspn(p, " \t\r\n");
             p[c] = '\0';
             
Index: src/buildus_main.c
===================================================================
RCS file: /cvsroot/roadmap/roadmap/src/buildus_main.c,v
retrieving revision 1.5
diff -u -3 -p -u -r1.5 buildus_main.c
--- src/buildus_main.c	8 Nov 2004 07:51:39 -0000	1.5
+++ src/buildus_main.c	22 Nov 2004 07:23:25 -0000
@@ -95,7 +95,7 @@ static void buildus_scan_cities (int fip
 
    cities = roadmap_dictionary_open ("city");
 
-   if (! cities) return; // May not exist is all map files.
+   if (! cities) return; /* May not exist is all map files. */
 
    us_cities = buildmap_dictionary_open ("city");
 
Index: src/roadmap_address.c
===================================================================
RCS file: /cvsroot/roadmap/roadmap/src/roadmap_address.c,v
retrieving revision 1.12
diff -u -3 -p -u -r1.12 roadmap_address.c
--- src/roadmap_address.c	5 Jul 2004 18:49:07 -0000	1.12
+++ src/roadmap_address.c	22 Nov 2004 07:23:25 -0000
@@ -33,7 +33,7 @@
 #include "roadmap_math.h"
 #include "roadmap_config.h"
 #include "roadmap_gui.h"
-// #include "roadmap_street.h"
+/* #include "roadmap_street.h" */
 #include "roadmap_history.h"
 #include "roadmap_locator.h"
 #include "roadmap_trip.h"
@@ -193,7 +193,7 @@ static void roadmap_address_ok (const ch
    state         = (char *) roadmap_dialog_get_data ("Address", "State:");
 
    if (context->use_zip) {
-      return; // TBD: how to select by ZIP ? Need one more table in usdir.
+      return; /* TBD: how to select by ZIP ? Need one more table in usdir. */
    }
 
    city = (char *) roadmap_dialog_get_data ("Address", "City:");
Index: src/roadmap_download.c
===================================================================
RCS file: /cvsroot/roadmap/roadmap/src/roadmap_download.c,v
retrieving revision 1.10
diff -u -3 -p -u -r1.10 roadmap_download.c
--- src/roadmap_download.c	2 Sep 2003 00:33:45 -0000	1.10
+++ src/roadmap_download.c	22 Nov 2004 07:23:26 -0000
@@ -105,7 +105,7 @@ static RoadMapDownloadEvent RoadMapDownl
 
 static int roadmap_download_request (int size) {
 
-   // TBD: for the time being, answer everything is fine.
+   /* TBD: for the time being, answer everything is fine. */
 
    RoadMapDownloadCurrentFileSize = size;
    RoadMapDownloadDownloaded = 0;
Index: src/roadmap_history.c
===================================================================
RCS file: /cvsroot/roadmap/roadmap/src/roadmap_history.c,v
retrieving revision 1.8
diff -u -3 -p -u -r1.8 roadmap_history.c
--- src/roadmap_history.c	13 Jun 2004 18:04:08 -0000	1.8
+++ src/roadmap_history.c	22 Nov 2004 07:23:26 -0000
@@ -232,7 +232,7 @@ void roadmap_history_load (void) {
    }
 
    loaded = 1;
-   // RoadMapHistoryChanged = changed;
+   /* RoadMapHistoryChanged = changed; */
    RoadMapHistoryChanged = 0;
 }
 
Index: src/roadmap_nmea.c
===================================================================
RCS file: /cvsroot/roadmap/roadmap/src/roadmap_nmea.c,v
retrieving revision 1.5
diff -u -3 -p -u -r1.5 roadmap_nmea.c
--- src/roadmap_nmea.c	14 Nov 2004 18:52:42 -0000	1.5
+++ src/roadmap_nmea.c	22 Nov 2004 07:23:26 -0000
@@ -200,7 +200,7 @@ static int roadmap_nmea_gprmc (int argc,
 
 static int roadmap_nmea_gpgga (int argc, char *argv[]) {
 
-   return 0; // TBD.
+   return 0; /* TBD. */
 }
 
 
Index: src/roadmap_sunrise.c
===================================================================
RCS file: /cvsroot/roadmap/roadmap/src/roadmap_sunrise.c,v
retrieving revision 1.2
diff -u -3 -p -u -r1.2 roadmap_sunrise.c
--- src/roadmap_sunrise.c	25 Dec 2003 23:26:20 -0000	1.2
+++ src/roadmap_sunrise.c	22 Nov 2004 07:23:26 -0000
@@ -43,7 +43,7 @@
 #define ROADMAP_SUNSET -1
 
 
-// returns an angle in range of 0 to (2 * PI)
+/* returns an angle in range of 0 to (2 * PI) */
 
 static double roadmap_sunrise_getrange(double x) {
 
@@ -78,7 +78,7 @@ static time_t roadmap_sunrise_get_gmt(do
    temp2 = (int)(floor(temp1));
    temp1 = 60 * (temp1 - floor(temp1));
 
-   // fill in the tm structure
+   /* fill in the tm structure */
    curtime->tm_hour = temp2;
    curtime->tm_min = (int) temp1;
 
@@ -132,12 +132,12 @@ static time_t roadmap_sunrise_getriseors
    double roadmap_long = LU_TO_DEG(position->longitude);
 	double roadmap_lat  = LU_TO_DEG(position->latitude);
 
-	// not accurate for places too close to the poles
+	/* not accurate for places too close to the poles */
 	if(abs(roadmap_lat) > 63) {
        return -1;
 	}
 
-	// check for valid riseorset parameter
+	/* check for valid riseorset parameter */
 	if((riseorset!=1)&&(riseorset!=-1)) {
        return -1;
    }
@@ -147,7 +147,7 @@ static time_t roadmap_sunrise_getriseors
    curtime_gmt = *(gmtime(&gmt));
 
 
-	altitude = -0.833; //tbd: sets according to position->altitude.
+	altitude = -0.833; /* tbd: sets according to position->altitude. */
 
 	ephem2000day = 367.0 * (curtime.tm_year+1900)
                       - 7.0 * ((curtime.tm_year+1900)
@@ -192,7 +192,7 @@ static time_t roadmap_sunrise_getriseors
                   (utold - (gha + longitude + riseorset * correction));
    }
 
-	// utnew must now be converted into gmt.
+	/* utnew must now be converted into gmt. */
 
    result = roadmap_sunrise_get_gmt (utnew * DEGREES / 15, &curtime_gmt);
    if (result < gmt) {
@@ -326,5 +326,5 @@ int main(int argc, char **argv) {
    printf("sunset:  %d:%d\n", realtime->tm_hour, realtime->tm_min);
    return 0;
 }
-#endif // SUNRISE_PROGRAM
+#endif /* SUNRISE_PROGRAM */
 
Index: src/qt/roadmap_main.cc
===================================================================
RCS file: /cvsroot/roadmap/roadmap/src/qt/roadmap_main.cc,v
retrieving revision 1.7
diff -u -3 -p -u -r1.7 roadmap_main.cc
--- src/qt/roadmap_main.cc	6 Sep 2003 15:24:37 -0000	1.7
+++ src/qt/roadmap_main.cc	22 Nov 2004 07:23:27 -0000
@@ -104,14 +104,15 @@ void roadmap_main_add_tool_space(void) {
 }
 
 void roadmap_main_add_canvas(void) {
-// The canvas is implicitely added to the main window.
+/* The canvas is implicitely added to the main window.
 //	if (mainWindow) {
 //		mainWindow->addCanvas();
 //	}
+*/
 }
 
 void roadmap_main_add_status(void) {
-	// nothing to be done
+	/* nothing to be done */
 }
 
 void roadmap_main_show(void) {
@@ -156,7 +157,7 @@ void roadmap_main_set_status(const char 
 
 
 void roadmap_main_toggle_full_screen (void) {
-	// Not yet implemented (how to do this ??)
+	/* Not yet implemented (how to do this ??) */
 }
 
 
Index: src/roadmap_start.c
===================================================================
RCS file: /cvsroot/roadmap/roadmap/src/roadmap_start.c,v
retrieving revision 1.57
diff -u -3 -p -u -r1.57 roadmap_start.c
--- src/roadmap_start.c	5 Oct 2004 08:24:42 -0000	1.57
+++ src/roadmap_start.c	22 Nov 2004 07:23:26 -0000
@@ -231,8 +231,8 @@ static void roadmap_start_toggle_downloa
 
       if (! ProtocolInitialized) {
 
-         // roadmap_plugin_load_all
-         //      ("download", roadmap_download_subscribe_protocol);
+         /* roadmap_plugin_load_all */
+         /*      ("download", roadmap_download_subscribe_protocol); */
 
          roadmap_copy_init (roadmap_download_subscribe_protocol);
          roadmap_httpcopy_init (roadmap_download_subscribe_protocol);
@@ -551,15 +551,15 @@ static char const *RoadMapStartKeyBindin
    "-"               ROADMAP_MAPPED_TO "zoomout",
    "A"               ROADMAP_MAPPED_TO "address",
    "B"               ROADMAP_MAPPED_TO "returntrip",
-   // C Unused.
+   /* C Unused. */
    "D"               ROADMAP_MAPPED_TO "destination",
    "E"               ROADMAP_MAPPED_TO "deletemaps",
    "F"               ROADMAP_MAPPED_TO "full",
    "G"               ROADMAP_MAPPED_TO "gps",
    "H"               ROADMAP_MAPPED_TO "hold",
    "I"               ROADMAP_MAPPED_TO "intersection",
-   // J Unused.
-   // K Unused.
+   /* J Unused. */
+   /* K Unused. */
    "L"               ROADMAP_MAPPED_TO "location",
    "M"               ROADMAP_MAPPED_TO "mapdownload",
    "N"               ROADMAP_MAPPED_TO "newtrip",
@@ -568,13 +568,13 @@ static char const *RoadMapStartKeyBindin
    "Q"               ROADMAP_MAPPED_TO "quit",
    "R"               ROADMAP_MAPPED_TO "zoom1",
    "S"               ROADMAP_MAPPED_TO "starttrip",
-   // T Unused.
+   /* T Unused. */
    "U"               ROADMAP_MAPPED_TO "gpsnorthup",
-   // V Unused.
+   /* V Unused. */
    "W"               ROADMAP_MAPPED_TO "addaswaypoint",
    "X"               ROADMAP_MAPPED_TO "intersection",
-   // Y Unused.
-   // Z Unused.
+   /* Y Unused. */
+   /* Z Unused. */
    NULL
 };
 
@@ -747,8 +747,8 @@ void roadmap_start_unfreeze (void) {
 void roadmap_start (int argc, char **argv) {
 
 #ifdef ROADMAP_DEBUG_HEAP
-   // Do not forget to set the trace file using the env. variable MALLOC_TRACE,
-   // then use the mtrace tool to analyze the output.
+   /* Do not forget to set the trace file using the env. variable MALLOC_TRACE,
+    * then use the mtrace tool to analyze the output. */
    mtrace();
 #endif
 
